### PR TITLE
Akeyless plugins

### PR DESCRIPTION
**Pull request by d-goro**
adds Akeyless plugins

**Pull Request check list**

- [ ] Commit conforms to CONTRIBUTING.md?
- [ ] Proper tests/regressions included?
- [ ] Documentation updated?

**Affected functionality**

**Description of change**

**Which issue this PR fixes**

### CODE REVIEW

**Comment by evan2645** [comment]
Hey @d-goro , thank you very much for this contribution and sorry for the delay on our side.

The TL;DR here is we're not quite sure if/how we can accept this, and I wrote a new issue in response to that: https://github.com/spiffe/spire/issues/4163

We need to come up with a consistent policy, and answer questions around how to test and maintain these kinds of plugins. I'm afraid that it might take some time to figure that out, and I don't want to hold this PR up indefinitely. My recommendation would be to publish these plugins in the akeyless GH org instead, as an [external plugin](https://github.com/spiffe/spire-plugin-sdk/tree/main#overview) that folks can download and run. Your help in navigating #4163 would also be great if you're motivated to do so 😅

**Comment by d-goro** [comment]
> Hey @d-goro , thank you very much for this contribution and sorry for the delay on our side.
> 
> The TL;DR here is we're not quite sure if/how we can accept this, and I wrote a new issue in response to that: #4163
> 
> We need to come up with a consistent policy, and answer questions around how to test and maintain these kinds of plugins. I'm afraid that it might take some time to figure that out, and I don't want to hold this PR up indefinitely. My recommendation would be to publish these plugins in the akeyless GH org instead, as an [external plugin](https://github.com/spiffe/spire-plugin-sdk/tree/main#overview) that folks can download and run. Your help in navigating #4163 would also be great if you're motivated to do so 😅

Hey @evan2645 , thank you for answer. Could you point me where I can find manual regarding **external** plugin configuration? I looked [here](https://spiffe.io/docs/latest/deploying/spire_server/#server-configuration-file) but it has only information on how to configure **internal** plugins.

I am looking for explanation on what's the best practice to publish external plugin (binary?, github repo?) and how to configure it in spire system.

Thank you.

**Comment by azdagron** [comment]
Hey @d-goro, looks like that website is a little out of date. The freshest version of the docs is found inside the repo, [here](https://github.com/spiffe/spire/blob/main/doc/spire_server.md#plugin-configuration). The difference in configuration is inclusion of a `plugin_cmd` configurable that is a path to the binary and optionally (but recommended) a `plugin_checksum` configurable with the sha256 hash of the binary.

To author an external plugin, the only difference is that you provide a main function that invokes the pluginmain.Serve. The plugin sdk repo has a template demonstrating this [here](https://github.com/spiffe/spire-plugin-sdk/blob/b59b964f004ecdba7fa4f8daa9787229fad8b3a5/templates/agent/nodeattestor/nodeattestor.go#L121).

**Comment by evan2645** [comment]
Hey @d-goro .. just checking in here to see if you found everything you need. An external plugin would live in a standalone repo, so my recommendation is to publish it into @akeylesslabs or @akeyless-community. You can create GH actions there to build and publish the plugin as a binary that can be loaded into SPIRE via the config that @azdagron referenced

Since it seems like you're going in that direction, I'll go ahead and close this out for now. If you have any further questions, please feel free to ping back in an issue or on the [slack](https://slack.spiffe.io) ... and as mentioned, you can follow along #4163 - your participation there would be great!
